main.go

```diff
@@ -58,13 +63,17 @@ func main() {
 	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
-	opts := zap.Options{
-		Development: true,
-	}
+	var opts zap.Options
 	opts.BindFlags(flag.CommandLine)
 	flag.Parse()
 
-	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+	operatorCfg := config.Parse()
+	
+	ctrl.SetLogger(zap.New(
+		zap.UseFlagOptions(&opts),
+		zap.UseDevMode(operatorCfg.DevMode),
+		zap.Level(mustGetLogLevel(operatorCfg.LogLevel)),
```

Comment:
Another option could be avoided is to have a panic-killing app with status code 1. What do you think

Reply:
A panic exits with non-zero:

```
❯ cat /tmp/panic.go
package main

func main() {
    panic("oh no")
}
❯ go run /tmp/panic.go
panic: oh no

goroutine 1 [running]:
main.main()
	/tmp/panic.go:4 +0x39
exit status 2
```
Do you think we should push specifically for status code 1?

Reply:
As we talked, the panic returns the message feedback for user, so I consider ok.